lib/diff-plugin-outputs.js

```diff
@@ -6,197 +6,198 @@ const Zip = require('node-zip');
+    // export with current plugin script
+    exportFixtures(
+      currentOut,
+      path.join(__dirname, '..', scriptName),
+      path.join(__dirname, '..')
     );
-  }
-  catch (e) {
-    console.error(colors.red('[Error]') + ` Failed downloading compare plugin script: ${e.message}`);
-    process.exit(1);
-  }
-
-  // unzip archive
-  const unzip = new Zip(fs.readFileSync(compareArchive));
-  const unzipDir = path.join(tempDir, 'compareFiles');
-  fs.mkdirSync(unzipDir);
-  for (let file of Object.keys(unzip.files)) {
-    file = unzip.files[file];
-    const filePath = path.join(unzipDir, file.name);
-
-    if (file.dir) {
-      fs.mkdirSync(filePath);
+  
+    // get compare script and fixture files as archive
+    const compareArchive = path.join(tempDir, 'compare.zip');
+    try {
+      childProcess.execSync(
+        `git archive ${ref} -o ${compareArchive}`,
+        {
+          cwd: path.join(__dirname, '..'),
+          encoding: 'utf-8'
+        }
+      );
     }
-    else {
-      fs.writeFileSync(filePath, file.asNodeBuffer());
+    catch (e) {
+      console.error(colors.red('[Error]') + ` Failed downloading compare plugin script: ${e.message}`);
+      process.exit(1);
     }
-  }
-  del.sync(compareArchive, {force: true});
-
-  // export with compare plugin script
-  exportFixtures(
-    compareOut,
-    path.join(unzipDir, scriptName),
-    unzipDir
-  );
-
-  // make recursive diff
-  diffDir('');
-  findAddedFiles('');
-  console.log('Done.');
-
-  if (resolve) {
-    resolve(outputData);
-  }
-
-  // diffs recursively compare output vs. current output
-  function diffDir(subdir) {
-    const dir = path.join(compareOut, subdir);
-    const files = fs.readdirSync(dir);
-
-    for (const file of files) {
-      const filePath = path.join(subdir, file);
-      const filePathCompare = path.join(compareOut, filePath);
-      const filePathCurrent = path.join(currentOut, filePath);
-
-      if (!fs.existsSync(filePathCurrent)) {
-        console.log(colors.red(`File or directory ${filePath} was removed.`));
-        outputData.removedFiles.push(filePath);
+  
+    // unzip archive
+    const unzip = new Zip(fs.readFileSync(compareArchive));
+    const unzipDir = path.join(tempDir, 'compareFiles');
+    fs.mkdirSync(unzipDir);
+    for (let file of Object.keys(unzip.files)) {
+      file = unzip.files[file];
+      const filePath = path.join(unzipDir, file.name);
+  
+      if (file.dir) {
+        fs.mkdirSync(filePath);
       }
       else {
-        if (fs.lstatSync(filePathCompare).isDirectory()) {
-          diffDir(filePath);
+        fs.writeFileSync(filePath, file.asNodeBuffer());
+      }
+    }
+    del.sync(compareArchive, {force: true});
+  
+    // export with compare plugin script
+    exportFixtures(
+      compareOut,
+      path.join(unzipDir, scriptName),
+      unzipDir
+    );
+  
+    // make recursive diff
+    diffDir('');
+    findAddedFiles('');
+    console.log('Done.');
+  
+    if (resolve) {
```

Comment:
Is `resolve` not always available?

Anyway, why do you use a promise at all? All the functions are synchronous, so there is no need for an asynchronous response of the function.